main.py

```diff
@@ -148,7 +152,40 @@ def setup_target_task_training(args, target_tasks, model, strict):
                     args.allow_untrained_encoder_parameters, "No best checkpoint found to evaluate."
                 )
                 log.warning("Using untrained encoder parameters!")
-    return task_names_to_avoid_loading
+
+        # The version of the model before target task training for the new task.
+        pre_target_train_model = copy.deepcopy(model)
```

Comment:
This looks expensive, and it might increase GPU memory usage. Why the deep copy?

Reply:
I think it's cleanest to just make sure there's a checkpoint we can restore to, and passing around a path to that checkpoint.